test/integration/zk-discovery.bats

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bats
+
+load helpers
+
+ZK_HOST=127.0.0.1:2181
+
+function start_zk() {
+	docker run --name zk -p $ZK_HOST:2181 -d jplock/zookeeper
+}
+
+function stop_zk() {
+	docker rm -f -v zk > /dev/null
+}
+
+function teardown() {
+	stop_swarm_agents
+	stop_manager
+	stop_docker
+	stop_zk
+}
+
+@test "zk discovery should be working correctly" {
+	start_zk
+	start_docker 3
+
+	swarm_manage zk://${ZK_HOST}/test
+	swarm_join   zk://${ZK_HOST}/test
+
+ 	sleep 10 # wait for all nodes to join
```

Comment:
And if we must sleep, maybe we could do it as short as possible, like we do for `wait_until_reachable`: https://github.com/chanwit/swarm/blob/test-zk-discovery/test/integration/helpers.bash#L29